### PR TITLE
fix: Prevent triggering user details from click on self mention (WEBAPP-5547)

### PR DESCRIPTION
**Pull request by gregor**

### CODE REVIEW

File: app/script/user/UserRepository.js [hunk 1]
```diff
@@ -681,13 +681,17 @@ z.user.UserRepository = class UserRepository {
...
+   * @param {string} userId - User ID
    * @returns {Promise<z.entity.User>} Resolves with the matching user entity
    */
-  findUserById(user_id) {
-    for (const user_et of this.users()) {
-      if (user_et.id === user_id) {
-        return Promise.resolve(user_et);
+  findUserById(userId) {
+    if (!userId) {
+      return Promise.reject(new z.user.UserError(z.user.UserError.TYPE.MISSING_PARAMETER));
```
Comment (gregor) [inline app/script/user/UserRepository.js]: Previously any requestor of `get_user_by_id` would have to handle undefined as a potential return value. This should never have been the case.

File: app/script/view_model/content/MessageListViewModel.js [hunk 2]
```diff
@@ -613,7 +613,9 @@ z.viewModel.content.MessageListViewModel = class MessageListViewModel {
   handleClickOnMessage(messageEntity, event) {
     const hasMentions = messageEntity.mentions().length;
     const mentionElement = hasMentions && event.target.closest('.message-mention');
-    if (mentionElement) {
+    const userId = mentionElement.dataset.userId;
+
+    if (mentionElement && userId) {
```
Comment (gregor) [inline app/script/view_model/content/MessageListViewModel.js]: A self mention does not contain a user id. The click should not trigger the show user details flow.

File: app/script/user/UserRepository.js [hunk 3]
```diff
@@ -681,13 +681,17 @@ z.user.UserRepository = class UserRepository {
...
+    }
+
+    for (const userEntity of this.users()) {
```
Comment (atomrc) [inline app/script/user/UserRepository.js]: How about using `array.find` for that use case?

Comment (gregor) [reply app/script/user/UserRepository.js]: Ah. Yes that is definitely possible these day 🙈

File: app/script/view_model/content/MessageListViewModel.js [hunk 4]
```diff
@@ -613,7 +613,9 @@ z.viewModel.content.MessageListViewModel = class MessageListViewModel {
   handleClickOnMessage(messageEntity, event) {
     const hasMentions = messageEntity.mentions().length;
     const mentionElement = hasMentions && event.target.closest('.message-mention');
-    if (mentionElement) {
+    const userId = mentionElement.dataset.userId;
```
Comment (atomrc) [inline app/script/view_model/content/MessageListViewModel.js]: Will crash if `mentionElement` is not found/defined/null..

Comment (atomrc) [reply app/script/view_model/content/MessageListViewModel.js]: Should probably be moved inside the `if (mentionElement)` condition

File: app/script/view_model/content/MessageListViewModel.js [hunk 5]
```diff
@@ -613,7 +613,9 @@ z.viewModel.content.MessageListViewModel = class MessageListViewModel {
   handleClickOnMessage(messageEntity, event) {
     const hasMentions = messageEntity.mentions().length;
     const mentionElement = hasMentions && event.target.closest('.message-mention');
-    if (mentionElement) {
+    const userId = mentionElement && mentionElement.dataset.userId;
+
+    if (userId) {
       this.userRepository
         .get_user_by_id(mentionElement.dataset.userId)
```
Comment (atomrc) [inline app/script/view_model/content/MessageListViewModel.js]: And then you can use `userId` here

Comment (gregor) [reply app/script/view_model/content/MessageListViewModel.js]: Obviously...

**Pull request merged**